runtime/parachains/src/parathread_assigner.rs

```diff
@@ -0,0 +1,211 @@
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		BaseSpotPriceSet { amount: BalanceOf<T> },
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// Create a single parathread core order.
+		///
+		/// Parameters:
+		/// - `origin`:
+		///
+		/// Errors:
+		/// Events:
+		/// TODO weights
+		#[pallet::call_index(0)]
+		#[pallet::weight(1_000)]
+		pub fn place_order(origin: OriginFor<T>) -> DispatchResult {
+			let _res = ensure_signed(origin);
+			Ok(())
+		}
+
+		/// Set the base spot price. Should only be available to the runtime's SetPriceOrigin.
+		///
+		/// Parameters:
+		/// - `origin`: Must be root
+		/// - `amount`: The price to set as the base
+		///
+		/// Errors:
+		/// - `BadOrigin`
+		/// Events:
+		/// - `BaseSpotPriceSet(amount)`
+		/// TODO weights
+		#[pallet::call_index(1)]
+		#[pallet::weight(1_000)]
+		pub fn set_base_spot_price(origin: OriginFor<T>, amount: BalanceOf<T>) -> DispatchResult {
+			// Ensure the proper origin authority
+			T::SetPriceOrigin::ensure_origin(origin)?;
+
+			BaseSpotPrice::<T>::set(amount);
+			Self::deposit_event(Event::BaseSpotPriceSet { amount });
+			Ok(())
+		}
+	}
+
+	impl<T: Config> Pallet<T> {
+		/// Called by the initializer to initialize the pallet.
+		pub(crate) fn initializer_initialize(now: T::BlockNumber) -> Weight {
+			Weight::zero()
+		}
+
+		/// Called by the initializer to finalize the pallet.
+		pub(crate) fn initializer_finalize() {}
+
+		/// Called by the initializer to note that a new session has started.
+		pub(crate) fn initializer_on_new_session(
+			notification: &SessionChangeNotification<T::BlockNumber>,
+		) {
+		}
+
+		/// The spot price multiplier. This is based on the transaction fee calculations defined in:
+		/// https://research.web3.foundation/en/latest/polkadot/overview/2-token-economics.html#setting-transaction-fees
+		///
+		/// Returns:
+		/// - A `FixedI64` in the range of 1.0 - `FixedI64::MAX`
+		/// Parameters:
+		/// - `traffic`: The previously calculated multiplier, can never go below 1.0.
+		/// - `queue_capacity`: The max size of the order book.
+		/// - `queue_size`: How many orders are currently in the order book.
+		/// - `target_queue_utilization`: How much of the queue_capacity should be ideally occupied, expressed in percentages(perbill).
+		/// - `variability`: A variability factor, i.e. how quickly the spot price adjusts. This number can be chosen by
+		///                  p/(k*(1-s)) where p is the desired ratio increase in spot price over k number of blocks.
+		///                  s is the target_queue_utilization. A concrete example: v = 0.05/(20*(1-0.25)) = 0.0033.
+		pub(crate) fn calculate_spot_price(
+			traffic: FixedI64,
+			queue_capacity: u32,
+			queue_size: u32,
+			target_queue_utilization: Perbill,
+			variability: Perbill,
+		) -> Option<FixedI64> {
```

Comment:
Consider using the [?](https://doc.rust-lang.org/reference/expressions/operator-expr.html#the-question-mark-operator)  operator. Preferably turning `Option` into `Result` for better reporting in case of error.

Reply:
Thanks, the if let chain was bothering me. Fixed.

Reply:
Did you maybe forget to commit and push? Still looks the same to me.